python/cuml/tests/test_hdbscan.py

```diff
@@ -132,16 +133,13 @@ def assert_condensed_trees(sk_agg, min_cluster_size):
 
 
 def assert_all_points_membership_vectors(cu_vecs, sk_vecs):
-    i = 0
-    while i < sk_vecs.shape[0]:
-        total_nz = 0
-        for j in range(sk_vecs.shape[1]):
-            if sk_vecs[i][j] < 1e-3 or cu_vecs[i][j] < 1e-3:
-                break
-            total_nz += 1
-        if total_nz == sk_vecs.shape[1]:
-            assert np.allclose(cu_vecs[i], sk_vecs[i], atol=0.1, rtol=0.1)
-        i += 1
+    if sk_vecs.shape == cu_vecs.shape:
+        cu_labels_sorted = np.argsort(cu_vecs)
+        sk_labels_sorted = np.argsort(sk_vecs)
+
+        for i in range(sk_labels_sorted.shape[1]):
+            assert adjusted_rand_score(cu_labels_sorted[:, i],
+                                       sk_labels_sorted[:, i]) >= 0.90
```

Comment:
I recall you mentioning the first k (where k is some small value <5-ish) closest labels consistently have an adjusted rand score of >= 0.99. Can we add a distinction here to use a higher threshold when k <= that value?

Reply:
Does this one still need to be addressed?